backend/src/controllers/productControllers.js

```diff
@@ -1,6 +1,12 @@
 const Product = require("../models/productModel");
 const Shop = require("../models/shopModel");
-const logger = require("../../logger");
+const initializeLogger = require("../../logger");
+
+let logger;
+
+initializeLogger().then((initializedLogger) => {
+  logger = initializedLogger;
+});
```

Comment:
Not really satisfied with this implementation for initializing the logger every time it is needed, but couldn't think of any good workaround as of yet.  Feel free to suggest ways to refactor this.